### PR TITLE
Copy to or from a multisampled texture should be allowed

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Comment by Richard-Yunchao** [comment]
WebGPU spec says:
  - The imageCopyTexture subresource size of imageCopyTexture is equal to copySize if either of the following conditions is true:
    - imageCopyTexture.texture.[[descriptor]].format is a depth-stencil format.
    - imageCopyTexture.texture.[[descriptor]].sampleCount is greater than 1.

So I think the current test is incorrect. This is the fix. In addition, Dawn's implementation is incorrect too. Per my understanding about the spec, we should support full copy for multisampled texture. But I didn't do any investigation about copy multisampled texture in Dawn project, please let me know if there are any known difficulties to do that @Kangz , @austinEng . Otherwise, I will fix it in Dawn project.

File: src/webgpu/api/validation/image_copy/texture_related.spec.ts
Comment (Kangz) [inline src/webgpu/api/validation/image_copy/texture_related.spec.ts:92]: nit: "Test that ..." or "Check validation of ..."

File: src/webgpu/api/validation/image_copy/texture_related.spec.ts
Comment (Kangz) [inline src/webgpu/api/validation/image_copy/texture_related.spec.ts:98]: nit: we could have something like `copyWidthModifier: [0, -1]` and same for height to check exactly the condition.

File: src/webgpu/api/validation/image_copy/texture_related.spec.ts
Comment (Kangz) [inline src/webgpu/api/validation/image_copy/texture_related.spec.ts:95]: Copying multisampled textures is only supported for copyT2T, not copyB2T and copyT2B

**Review by Kangz — APPROVED** [review]
I think Dawn already implements the correct validation, see `ValidateImageCopyTexture`

File: src/webgpu/api/validation/image_copy/texture_related.spec.ts
Comment (Jiawei-Shao) [reply src/webgpu/api/validation/image_copy/texture_related.spec.ts:95]: The validation tests on T2T multisampled texture copies are here:
https://github.com/gpuweb/cts/blob/main/src/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.ts

The operation tests on T2T multisampled texture copies haven't been added yet. Tracking Dawn issue: https://bugs.chromium.org/p/dawn/issues/detail?id=1033

**Review by kainino0x — COMMENTED** [review]
The image_copy directory only tests B2T/T2B/writeTexture. None of these allow copying multisampled textures at all, even if they cover the full subresource (only T2T allows that). So this change doesn't seem right to me.

**Review by kainino0x — CHANGES_REQUESTED** [review]
^

**Comment by kainino0x** [comment]
Note, in the spec, the rules are specified individually in copyBufferToTexture, copyTextureToBuffer, and writeTexture. We unfortunately don't have a common piece of validation text that describes it for all 3.

**Comment by Richard-Yunchao** [comment]
> Note, in the spec, the rules are specified individually in copyBufferToTexture, copyTextureToBuffer, and writeTexture. We unfortunately don't have a common piece of validation text that describes it for all 3.

Yeah. I noticed that too. I mentioned that I would close this PR at https://github.com/gpuweb/cts/pull/679 (please take a look at that PR if you have time, Kai). Alternatively, I can add some tests for formats: copy to/from depth/stencil formats requires that it should be a full copy, just as compressed formats for T2T copy. In addition, there are more requirements for depth/stencil formats. 

And there should be more tests for copy validation for 2D texture. I mean, 2D texture copy validation is not complete, I will revisit/review it too.

**Comment by kainino0x** [comment]
Oh, I didn't realize this was to be closed. We should just close it now if we already know we're going to close it.

I wasn't going to review #679 since Corentin already reviewed it, but it's short so I can.

**Comment by Richard-Yunchao** [comment]
> Oh, I didn't realize this was to be closed. We should just close it now if we already know we're going to close it.
> 
> I wasn't going to review #679 since Corentin already reviewed it, but it's short so I can.

Ah... I should have closed it. Thanks for your input about this PR, Corentin, Jiawei and Kai. 

For #679, should I merge it by myself? More broadly, I feel like that the criteria of landing a PR in CTS (or all PRs on on github) is not very clear. So I always wait you folks (In most cases it is Kai, sometimes it is Corentin) to merge my PR. And I myself will not merge it even someone give a LGTM/approval. 

For Dawn project, if someone gives a +2, then I will merge the patch by myself, with some tiny fixes for nits if needed. However, github doesn't have +2. And LGTM/approval for github PR is more like a +1 in Dawn project, per my understanding. 

WDYT, Kai? Should I merge #679 and other later PRs by myself now that Corentin has given an approval?

**Comment by kainino0x** [comment]
I've been thinking we should have a better CR flag policy for this repo. For now, here's how I've been thinking about it:
"Approval" means the change would be OK to land with any errors addressed. The PR author should request another review if the changes are very substantial. If not, it's fine to land them (as long as the checklist items look good - OK to ignore "reviewer" checkboxes if they're not very relevant). Fixes can be made in new PRs.

**Comment by Richard-Yunchao** [comment]
> I've been thinking we should have a better CR flag policy for this repo. For now, here's how I've been thinking about it:
> "Approval" means the change would be OK to land with any errors addressed. The PR author should request another review if the changes are very substantial. If not, it's fine to land them (as long as the checklist items look good - OK to ignore "reviewer" checkboxes if they're not very relevant). Fixes can be made in new PRs.

I noted your review at #679. But the problem about criteria of landing a PR in CTS is still there. 

I like yours suggestion. I guess that I can follow it in this way if someone gives an approval but doesn't merge it: 
  - for small PR, I will merge it directly if everything is fine, 
  - for substantial PR, I will ask a second review, 
  - for a PR that is somewhere in between, I may ask something like "Can you merge the PR if it is fine? Otherwise I will merge it tomorrow".  

Just like you said, if there are bugs, we can fix them in new a PR soon. I am not a random contributor who submits a PR and then disappears, :).

**Pull request closed without merging**
